pkg/apis/datadoghq/v1alpha1/datadogagent_types.go

```diff
@@ -167,7 +167,7 @@ type DatadogAgentSpecAgentSpec struct {
 	// Allow to put custom configuration for the agent, corresponding to the datadog.yaml config file
 	// See https://docs.datadoghq.com/agent/guide/agent-configuration-files/?tab=agentv6 for more details.
 	// +optional
-	CustomConfig string `json:"customConfig,omitempty"`
+	CustomConfig *CustomConfigSpec `json:"customConfig,omitempty"`
```

Comment:
I think we need to document the behaviour when both `ConfigData` and `ConfigMap` are set, and explain which field will be ignored

Reply:
it will not be ignore, but the DatadogAgent will be set as invalid